### PR TITLE
Edit categories layouts and fragment classes

### PR DESCRIPTION
**Pull request by jequeen**
Adding layouts for edit categories and the fragment classes for those layouts.

Not included:
- Adding categories functionality
- Deleting categories functionality
- Hiding categories functionality
- Re-ordering categories functionality

### CODE REVIEW

File: app/src/main/res/layout-sw600dp/category_edit_button.xml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        app:layout_constraintHorizontal_bias="0.0"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/top_view">
+
+        <com.willowtree.vocable.customviews.VocableImageButton
+            android:id="@+id/move_category_up_button"
+            android:layout_width="56dp"
+            android:layout_height="56dp"
+            android:layout_marginStart="16dp"
+            android:layout_marginEnd="8dp"
```
Comment (nicholas-cook) [inline app/src/main/res/layout-sw600dp/category_edit_button.xml:26]: Hardcoded values here and elsewhere

File: app/src/main/res/layout/category_edit_button.xml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            app:layout_constraintTop_toTopOf="@id/category_container"
+            app:layout_constraintBottom_toTopOf="@id/move_category_up_button"
+            app:layout_constraintStart_toStartOf="@id/move_category_up_button" />
+
+        <com.willowtree.vocable.customviews.VocableImageButton
+            android:id="@+id/move_category_up_button"
+            android:layout_width="56dp"
+            android:layout_height="56dp"
+            android:layout_marginStart="16dp"
+            android:layout_marginEnd="8dp"
```
Comment (nicholas-cook) [inline app/src/main/res/layout/category_edit_button.xml:38]: Hardcoded values here and elsewhere

File: app/src/main/res/layout-land/category_edit_button.xml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:id="@+id/parent_layout"
+    android:layout_width="0dp"
+    android:layout_height="0dp"
+    android:orientation="vertical"
+    android:theme="@style/SettingsTheme"
+    android:layout_rowWeight="1"
+    android:layout_columnWeight="1">
+
+    <com.willowtree.vocable.customviews.VocableConstraintLayout
```
[the remark this answers is not in the document]
Comment (nicholas-cook) [reply app/src/main/res/layout-land/category_edit_button.xml:12]: Same goes for the other versions of this file

File: app/src/main/java/com/willowtree/vocable/settings/EditCategoriesViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+import com.willowtree.vocable.room.Category
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import org.koin.core.inject
+
+class EditCategoriesViewModel(numbersCategoryId: String, mySayingsCategoryId: String) :
+    BaseViewModel(numbersCategoryId, mySayingsCategoryId) {
+
+    companion object {
+        private const val CATEGORY_UPDATED_DELAY = 2000L
```
[the remark this answers is not in the document]
Comment (CarolineLaw) [reply app/src/main/java/com/willowtree/vocable/settings/EditCategoriesViewModel.kt:16]: Yeah, it seems that this is also in `EditPhrasesViewModel` so we can also update it to be one value there too

File: app/src/main/res/layout-land/category_edit_button.xml [hunk 5]
```diff
@@ -1,81 +1,69 @@
...
     xmlns:app="http://schemas.android.com/apk/res-auto"
-    android:id="@+id/parent_layout"
+    android:id="@+id/category_container"
     android:layout_width="0dp"
-    android:layout_height="0dp"
-    android:orientation="vertical"
-    android:theme="@style/SettingsTheme"
-    android:layout_rowWeight="1"
-    android:layout_columnWeight="1">
+    android:layout_height="match_parent"
```
Comment (nicholas-cook) [inline app/src/main/res/layout-land/category_edit_button.xml:6]: So you might still need the `android:layout_rowWeight="1"` and `android:layout_columnWeight="1"` attributes from the original parent view. I'd run that and double check.

**Pull request merged**